### ISSUE TITLE
Adding an address question as an eligibility condition causes a vague error 

### ISSUE THREAD

**Issue by melanie-exygy** [civiform/civiform#5002]
**Describe the bug**
Adding a question of address type as an eligibility condition causes a generic error (500 status code) with the message "Oops, an error occurred."

**To Reproduce**

1. Open the staging instance of CiviForm.
2. As a CiviForm admin, go to the Program Dashboard.
3. Click the "Edit" button on a draft program (or create a draft).
4. If the program doesn't already have an address question, add one.  Make sure "address correction" is enabled.
5. Click the "Edit eligibility condition" button.
6. Select the checkbox next to the address question.
7. Click the "Add condition" button.
8. See the "Oops, an error occurred message."

**Expected behavior**
The address question can be added as an eligibility condition.  There is no error.
If either address correction or service area validation are turned off (as is the case in the staging environment), eligibility for address questions should be disabled, meaning that the address question does not appear in the list of potential conditions.

**Screenshots**
<img width="1440" alt="image" src="https://github.com/civiform/civiform/assets/134556723/0041f72f-0822-4221-a72c-dfb61ecb6c4f">

**Desktop (please complete the following information):**

- OS: iOS
- Browser Chrome
- Version 113.0.5672.126 

**Smartphone (please complete the following information):**

Did not verify on a smartphone.

**Comment by bion** [comment]
@melanie-exygy including the stack trace would be helpful. Can find in the AWS logs or by reproing locally.

**Comment by bion** [comment]
My guess is this is a result of staging not having configuration for ESRI

**Comment by melanie-exygy** [comment]
@bion Good idea about the stack trace.  I'll add it as soon as I figure out how to check the AWS logs.  The issue isn't reproducible locally.
As for this being caused by a missing ESRI configuration on staging, both @rockycodes and @nb1701 suspected the same thing.  Nonetheless, it would be good to find a graceful way to handle that situation in case the configurations ever get messed up in production.  Nick suggested handling the exception with an error toast message that tells you service area validation isn't configured correctly.  Rocky suggested eligibility for address questions should be disabled unless both address correction and service area validation are turned on.
